pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -542,10 +542,21 @@ func (ctrl csiSnapshotSideCarController) removeContentFinalizer(content *crdv1.V
 		// the finalizer does not exit, return directly
 		return nil
 	}
+	var patches []utils.PatchOp
 	contentClone := content.DeepCopy()
-	contentClone.ObjectMeta.Finalizers = utils.RemoveString(contentClone.ObjectMeta.Finalizers, utils.VolumeSnapshotContentFinalizer)
+	patchPathSourceFin := strings.ReplaceAll(utils.VolumeSnapshotAsSourceFinalizer, "/", "~1")
+	patches = append(patches,
+		utils.PatchOp{
+			Op:    "remove",
+			Path:  "/metadata/finalizers/0",
+			Value: patchPathSourceFin,
+		}, utils.PatchOp{
+			Op:    "replace",
+			Path:  "/metadata/finalizers",
+			Value: nil,
+		})
```

Comment:
I think this may do more than we want here. In this code, it looks like it will remove the first finalizers regardless of the value, and then replace all finalizers with `nil`. Should we instead pull the finalizers slice, remove the content finalizer, and then replace the entire slice with this new one?

Reply:
Updated the PR, PTAL @ggriffiths